fuel-core-interfaces/src/db.rs

```diff
@@ -918,3 +948,42 @@ pub mod helpers {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::db::helpers::DummyDb;
+    use crate::db::KvStoreError;
+    use fuel_storage::Storage;
+    use fuel_tx::Transaction;
+    use fuel_types::Bytes32;
+
+    #[test]
+    fn transactions_db() -> Result<(), KvStoreError> {
+        let mut db: Box<dyn Storage<Bytes32, Transaction, Error = KvStoreError>> =
+            Box::new(DummyDb::filled());
+
+        let value: Transaction = Transaction::default();
+        let key = value.id();
+
+        //before insert
+        assert!(!db.contains_key(&key)?);
```

Comment:
I would like to reuse all the tests below and only change the db, value and key for the specific implementation. That way, I could test remove some duplication.
 
But not entirely sure how to do it without macros or leveraging generics. Maybe you have a better suggestion.

Reply:
I'd recommend looking into [rstest](https://github.com/la10736/rstest) as we already use it in other areas. The db could be instantiated via fixture.

Reply:
Thanks for the info!

I added more test cases and removed most of the duplication, and also found a way to to instantiate the db without needing fixtures after all.